Util/FormatNegotiator.php

```diff
@@ -11,134 +11,45 @@
 
 namespace FOS\RestBundle\Util;
 
+use FOS\RestBundle\Negotiation\FormatNegotiator as BaseFormatNegotiator;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\RequestMatcherInterface;
-use Negotiation\FormatNegotiator as BaseFormatNegotiator;
-use Negotiation\AcceptHeader;
 
+/**
+ * @deprecated since 1.7, to be removed in 2.0. Use {@link \FOS\RestBundle\Negotiation\FormatNegotiator} instead.
+ */
 class FormatNegotiator implements MediaTypeNegotiatorInterface
 {
-    private $formatNegotiator;
-    private $map = array();
+    private $negotiator;
 
     public function __construct()
     {
-        $this->formatNegotiator = new BaseFormatNegotiator();
-    }
+        @trigger_error('FOS\RestBundle\Util\FormatNegotiator is deprecated since version 1.7 and will be removed in 2.0. Use FOS\RestBundle\Negotiation\FormatNegotiator instead.', E_USER_DEPRECATED);
```

Comment:
in such case, Symfony generally puts the warning outside the class declaration, to trigger it when the class is autoloaded